src/charm.py

```diff
@@ -454,6 +458,12 @@ def _common_install_code(self, state: CharmState) -> bool:  # noqa: C901
             logger.error("Failed to install charm dependencies")
             raise
 
+        try:
+            _setup_runner_manager_user()
+        except SubprocessError:
+            logger.error("Failed to setup runner manager user")
```

Comment:
I think we can maybe use `logger.exception` here?

Reply:
Thank you. The error will get re-raised in the line afterwards and will cause the charm to go into error state. I think the stack trace will be available in the logs then. This pattern is used also in the rest of the function (which might not be visible in the diff), so I wouldn't like to change this in this PR.